starter_code/elevator.js

```diff
@@ -1,19 +1,73 @@
+/*jshint esversion: 6 */
+
 class Elevator {
   constructor(){
     this.floor      = 0;
     this.MAXFLOOR   = 10;
+    this.waitingList = [];
+    this.passengers = [];
     this.requests   = [];
+    this.direction = "up";
+  }
+
+  start() {
+    let timer = setInterval(() => this.update(), 1000);
+    setTimeout(() => this.stop(timer), 10000);
```

Comment:
Better don't call stop inside start. This way, the elevator is always going to stop 10 seconds after starting although there are requests pending.